src/test/java/dog/pawbook/logic/parser/EditDogCommandParserTest.java

```diff
@@ -0,0 +1,213 @@
+
+        // invalid breed followed by valid dob
+        assertParseFailure(parser, "1" + INVALID_BREED_DESC + DATEOFBIRTH_DESC_ASHER, Breed.MESSAGE_CONSTRAINTS);
+
+        // valid breed followed by invalid breed. The test case for invalid breed followed by valid breed
+        // is tested at {@code parse_invalidValueFollowedByValidValue_success()}
+        assertParseFailure(parser, "1" + BREED_DESC_BELL + INVALID_BREED_DESC, Breed.MESSAGE_CONSTRAINTS);
+
+        // while parsing {@code PREFIX_TAG} alone will reset the tags of the {@code Owner} being edited,
+        // parsing it together with a valid tag results in error
+        assertParseFailure(parser, "1" + TAG_DESC_FRIENDLY + TAG_DESC_QUIET + TAG_EMPTY, Tag.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, "1" + TAG_DESC_FRIENDLY + TAG_EMPTY + TAG_DESC_QUIET, Tag.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, "1" + TAG_EMPTY + TAG_DESC_FRIENDLY + TAG_DESC_QUIET, Tag.MESSAGE_CONSTRAINTS);
+
+        // multiple invalid values, but only the first invalid value is captured
+        assertParseFailure(parser, "1" + INVALID_NAME_DESC + INVALID_BREED_DESC + VALID_DATEOFBIRTH_ASHER
+                        + VALID_SEX_BELL, Name.MESSAGE_CONSTRAINTS);
+    }
+
+    @Test
+    public void parse_allFieldsSpecified_success() {
+        Integer targetId = ID_SECOND_DOG;
+        String userInput = targetId + BREED_DESC_ASHER + TAG_DESC_FRIENDLY
+                + DATEOFBIRTH_DESC_ASHER + SEX_DESC_ASHER + NAME_DESC_ASHER + TAG_DESC_QUIET;
+
+        EditDogDescriptor descriptor = new EditDogDescriptorBuilder().withName(VALID_NAME_ASHER)
+                .withBreed(VALID_BREED_ASHER).withDob(VALID_DATEOFBIRTH_ASHER).withSex(VALID_SEX_ASHER)
+                .withTags(VALID_TAG_FRIENDLY, VALID_TAG_QUIET).build();
+        EditDogCommand expectedCommand = new EditDogCommand(targetId, descriptor);
+
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_someFieldsSpecified_success() {
+        Integer targetId = ID_FIRST_DOG;
+        String userInput = targetId + BREED_DESC_BELL + DATEOFBIRTH_DESC_ASHER;
+
+        EditDogDescriptor descriptor = new EditDogDescriptorBuilder().withBreed(VALID_BREED_BELL)
+                .withDob(VALID_DATEOFBIRTH_ASHER).build();
+        EditDogCommand expectedCommand = new EditDogCommand(targetId, descriptor);
+
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_oneFieldSpecified_success() {
+        // name
+        Integer targetId = ID_THIRD_DOG;
+        String userInput = targetId + NAME_DESC_ASHER;
+        EditDogDescriptor descriptor = new EditDogDescriptorBuilder().withName(VALID_NAME_ASHER).build();
+        EditDogCommand expectedCommand = new EditDogCommand(targetId, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
```

Comment:
it will be good to test against `EditCommandParser` as well. I know it is an abstract class, but it is possible to create a simple implementation and test against that, especially for things like tags which there is a protected method that handles it for all classes. That way things like preamble etc only need to be tested once and not repeated since having repeated test cases on the same thing does not help with code coverage or prevent bugs but only increase the execution time of tests.